packages/plugin-ext/src/main/browser/plugin-ext-deploy-command.ts

```diff
@@ -104,11 +104,18 @@ export class DeployQuickOpenItem extends QuickOpenItem {
         if (mode !== QuickOpenMode.OPEN) {
             return false;
         }
-        const promise = this.pluginServer.deploy(this.name);
-        promise.then(() => {
-            this.hostedPluginSupport.initPlugins();
+
+        const deployedPlugins = this.hostedPluginSupport.getDeployedPlugins();
+        deployedPlugins.then(async metadata => {
+            const ids = metadata.map(value => value.model.id);
+
+            await this.pluginServer.deploy(this.name);
+            const delta = (await this.hostedPluginSupport.getDeployedPlugins()).filter((value: PluginMetadata) => ids.indexOf(value.model.id) < 0);
+
+            this.hostedPluginSupport.initPlugins(delta);
```

Comment:
Does it mean that any client which calls `initPlugins` should make sure that it does not provide already loaded plugins? Maybe it should be handled by `HostedPluginSupport`?

What about hosted mode? Hosted plugin seems to be deployed each time.

Reply:
@evidolob Could you comment pls

Reply:
I think that @akosyakov are right, better to handle it `HostedPluginSupport`.
As for hosted plugin, it deployed same as regular plugin, we just fetch metha info from different source

Reply:
I think that @akosyakov are right, better to handle it `HostedPluginSupport`.
As for hosted plugin, it deployed same as regular plugin, we just fetch metha info from different source

Reply:
@akosyakov @evidolob 
Thank you for notice. I've updated the PR